core/src/lib.rs

```diff
@@ -293,38 +298,37 @@ impl Schema {
         self.get_entid(x).map(|e| self.is_attribute(e)).unwrap_or(false)
     }
 
-    pub fn as_edn_value(&self) -> edn::Value {
+    pub fn to_edn_value(&self) -> edn::Value {
         let mut all_values = Vec::new();
         for (entid, attribute) in &self.schema_map {
             let mut attribute_map: BTreeMap<edn::Value, edn::Value> = BTreeMap::default();
-            attribute_map.insert(edn::Value::NamespacedKeyword(NamespacedKeyword::new("db", "id")), edn::Value::Keyword(Keyword::new(entid.to_string())));
-
             let some_ident = self.get_ident(entid.clone());
             if some_ident.is_some() {
                 let ident = some_ident.unwrap();
-                attribute_map.insert(edn::Value::NamespacedKeyword(NamespacedKeyword::new("db", "ident")), edn::Value::NamespacedKeyword(ident.clone()));
+                attribute_map.insert(values::DB_IDENT.clone(), edn::Value::NamespacedKeyword(ident.clone()));
```

Comment:
Sorry, I wasn't clear, and now I realize I wasn't even correct.  `:db/id` is special, in that it's not really "part of the store", it's syntax for the map notation that Datomic supports.  (See http://docs.datomic.com/transactions.html and my about-to-land PR for #180 for more on the map notation.)  It doesn't assign an ident (like `:db/ident`), it (optionally) specifies an entid for the resulting map.  That is, it's assigning an internal identifier (entid) rather than an external identifier (ident).

Now, I think it would be correct to produce `{:db/id entid}` all the time, but it ties the abstract symbolic representation that is all built on external identifiers (all the `values::*` are keywords -- idents -- not entids) that you produce in this function to the exact store (with the exact same internal identifiers -- entids).  It will be more useful to be able to transplant the representation across stores by not having the internal identifiers.

So: no `:db/id` at all, and a `:db/ident` only if there is a registered keyword.  Which is what I think you had!  Sorry to double back on you.